src/GoTrueClient.ts

```diff
@@ -2097,7 +2097,7 @@ export default class GoTrueClient {
       willUnref = true
     }
 
-    if(!willUnref && !GoTrueClient.memoryLeakWarningShown) { // prevents the warning from being spammed / shown multiple times
+    if(!willUnref && !GoTrueClient.memoryLeakWarningShown && (typeof isBrowser !== "function" || !isBrowser())) { // prevents the warning from being spammed / shown multiple times
       console.warn("[GoTrueClient] dereferencing `setInterval` is not supported in your JavaScript Runtime. This will cause memory leaks unless you call the `stopAutoRefresh` method before losing the last reference to a `GoTrueClient` instance. If you are using `supabase-js`, this can be accomplished by calling `supabase.auth.stopAutoRefresh()`. For more information, see https://github.com/supabase/gotrue-js/issues/856")
       GoTrueClient.memoryLeakWarningShown = true
```

Comment:
@hf this line is what causes the message to only be shown once, as it sets a static/class flag after the first showing. Also, this revised version uses the changes you suggested about `isBrowser`.